src/binaryen-c.cpp

```diff
@@ -1066,26 +1066,39 @@ BinaryenExpressionRef BinaryenGlobalSet(BinaryenModuleRef module,
   return static_cast<Expression*>(
     Builder(*(Module*)module).makeGlobalSet(name, (Expression*)value));
 }
+
+// All memory instructions should pass their memory name parameter through this
+// helper function. It maintains compatibility for when JS calls memory
+// instructions that don't specify a memory name (send null), by assuming the
+// singly defined memory is the intended one. This function takes in the memory
+// name passed to API functions to avoid duplicating the nullptr logic check in
+// each instruction
+static Name getMemoryName(BinaryenModuleRef module, const char* memName) {
+  const char* memoryName = memName;
+  if (memName == nullptr && module->memories.size() == 1) {
```

Comment:
```suggestion
static Name getMemoryName(BinaryenModuleRef module, const char* memoryName) {
  if (memoryName == nullptr && module->memories.size() == 1) {
```
The parameter can be reused.